Code/Framework/AzFramework/AzFramework/Entity/GameEntityContextComponent.cpp

```diff
@@ -20,6 +21,52 @@
 
 namespace AzFramework
 {
+    //=========================================================================
+    // Hierarchy Handling Utilities
+    //  - EraseValue
+    //  - EnsureChildList
+    //  - FindEntity
+    //=========================================================================
+    template<class T>
+    static bool EraseValue(AZStd::vector<T>& vec, const T& value)
+    {
+        // 1) "Remove": move all elements != value to the front.
+        auto it = AZStd::remove(vec.begin(), vec.end(), value);
+
+        // After this:
+        // - [vec.begin(), it)   contains the elements you KEEP, compacted
+        // - [it, vec.end())     contains now-meaningless "garbage" (old values)
+
+        // 2) If anything was actually "removed", trim the vector's size.
+        if (it != vec.end())
+        {
+            vec.erase(it, vec.end()); // physically shorten the vector
+            return true;              // we did remove at least one element
+        }
+        return false;                 // nothing matched 'value'
+    }
```

Comment:
instead of writing this function, use 
`AZStd::erase_if` for a predicate version
`AZStd::erase` for a non predicate version
 https://en.cppreference.com/w/cpp/container/vector/erase2
it is a c++20 addition that does the remove/erase idiom internally and returns the number of elements erased.